base/stream.jl

```diff
@@ -240,6 +253,63 @@ function init_stdio(handle::Ptr{Cvoid})
     end
 end
 
+"""
+    open(fd::OS_HANDLE) -> IO
+
+Take a raw file descriptor wrap it in a Julia-aware IO type,
+and take ownership of the fd handle.
+Call `open(Libc.dup(fd))` to avoid the ownership capture
+of the original handle.
+
+WARNING: do not call this on a handle that's already owned by
```

Comment:
You could make this a Documenter-style warning:
```suggestion
!!! warn
    Do not call this on a handle that's already owned by some
    other part of the system.
```